core/src/main/scala/io/finch/EndpointResult.scala

```diff
@@ -60,69 +46,49 @@ sealed abstract class EndpointResult[+A] {
     case _ => None
   }
 
-  /**
-   * Awaits for an [[Output]] wrapped with [[Try]] (indicating if the [[com.twitter.util.Future]]
-   * is failed).
-   *
-   * @note This method is blocking. Never use it in production.
-   *
-   * @return `Some(output)` if this endpoint was matched on a given input, `None` otherwise.
-   */
-  final def awaitOutput(d: Duration = Duration.Top): Option[Try[Output[A]]] = this match {
-    case EndpointResult.Matched(_, _, out) => Some(Await.result(out.liftToTry.run, d))
+  def awaitOutput(d: Duration = Duration.Top)(implicit e: Effect[F]): Option[Try[Output[A]]] = this match {
+    case EndpointResult.Matched(_, _, out) =>
+      val promise = Promise[Try[Output[A]]]
```

Comment:
Is there a way to avoid bouncing through `Promise` here? I think `unsafeRunTimed` already does what we want.

```scala
Effect[F].toIO(out).unsafeRunTimed(durection).get
```